packages/client/src/types/common/metadata.types.ts

```diff
@@ -0,0 +1,3 @@
+export type Metadata = {
+  [k: string]: unknown;
```

Comment:
Can we be more explicit instead of using `unknown`? For example, with this definition, I could pass functions and objects, which should not be allowed. So perhaps this could be defined as:

```suggestion
  [k: string]: number | string | boolean;
```

I think these types are the ones that make the most sense to be used for metadata. What do you think?

Reply:
Well accordingly to the apispec, in fact this is an object that accepts anything that is supported in json.

Reply:
so you could possible add inside this, another objects, arrays, etc. So I agree to keep unknown here 🤔

Reply:
I see. I agree regarding objects and arrays but functions I think should not be allowed. Let me check if there is already a type that we can reuse for that.

Reply:
There is not one type readily available and there is a lot of [discussion](https://github.com/microsoft/TypeScript/issues/1897) about it but I tried this type definition and it worked:

```js

export type JSONValue =
  | null
  | boolean
  | number
  | string
  | JSONValue[]
  | {[key: string]: JSONValue};

export type Metadata = {
  [k: string]: JSONValue;
}
```

What do you think? If you adopt this solution do not forget to export the type `JSONValue` from the package as well (not the Metadata type as it is only an alias) and not only from the file.

Reply:
I like this approach but I separated `JSONValue` to a new file, because I believe in can be used in other situations and it would makes more sense on its own.

Reply:
I used that type before, it works for me :)

Reply:
Thank you!